### PR TITLE
[RAPTOR-5739] Applying Jesse's technique to python

### PR DESCRIPTION
**Pull request by 4dahalibut**
# This repository is public. Do not put here any private DataRobot or customer's data: code, datasets, model artifacts, .etc.

## Summary
In  https://github.com/datarobot/datarobot-user-models/pull/377 we started checking if values could be converted into floats, and sorting them by their float-equivalent values if they could. Here I use the same technique in python.

### CODE REVIEW

File: tests/drum/test_units.py [hunk 1]
```diff
@@ -700,3 +700,26 @@ def test_run_java_server_entry_point_succeed(self):
         # required to properly shutdown py4j Gateway
         pred._setup_py4j_client_connection()
         pred._stop_py4j()
+
+
+@pytest.mark.parametrize("data_dtype,label_dtype", [(int, int), (float, int)])
```
Comment (meltzg) [inline tests/drum/test_units.py:705]: how about `(int, float)`?

File: custom_model_runner/datarobot_drum/drum/utils.py [hunk 2]
```diff
@@ -304,3 +304,11 @@ def capture_R_traceback_if_errors(r_handler, logger):
                 "t_traceback_err": traceback_exc,
             }
         raise
+
+
+def stringify_and_intify_label(label):
+    if isinstance(label, float):
+        if hash(label) != hash(int(label)):
+            raise ValueError("Class labels must be integers or strings")
```
Comment (meltzg) [inline custom_model_runner/datarobot_drum/drum/utils.py:312]: why _must_ labels be ints or strings? I know it's likely not common, but what's the issue with supporting labels like `1.1/2.2`?

**Review by rvorobii — APPROVED** [review]
other than the points Greg has -- looks good

File: tests/drum/test_units.py (hunk 1 above)
Comment (4dahalibut) [reply tests/drum/test_units.py:705]: so we're going to just not allow support float labels.

File: custom_model_runner/datarobot_drum/drum/utils.py (hunk 2 above)
Comment (4dahalibut) [reply custom_model_runner/datarobot_drum/drum/utils.py:312]: because I don't like them and they're bad

Comment (jmbannon) [reply custom_model_runner/datarobot_drum/drum/utils.py:312]: I don't think we should restrict this. I could see some 'regression' datasets with a few unique target values that someone might want to try as multiclass. We support it in DR so we should support it in training models

**Review by jmbannon — CHANGES_REQUESTED** [review]
We should allow floats

File: custom_model_runner/datarobot_drum/drum/artifact_predictors/sklearn_predictor.py [hunk 3]
```diff
@@ -59,14 +60,9 @@ def predict(self, data, model, **kwargs):
...
-                    str(label) for label in self.class_labels
-                ):
-                    error_message = "Wrong class labels {}. Use class labels detected by sklearn model: {}".format(
-                        self.class_labels, model.classes_
-                    )
-                    raise DrumCommonException(error_message)
-                labels_to_use = model.classes_
+                labels_to_use = marshal_labels(
+                    expected_labels=list(self.class_labels), actual_labels=list(model.classes_)
+                )
```
Comment (jmbannon) [inline custom_model_runner/datarobot_drum/drum/artifact_predictors/sklearn_predictor.py:65]: I don't think we need to change labels_to_use here. This only covers sklearn models. The check/fix should be downstream from this

**Review by jmbannon — APPROVED** [review]

**Pull request merged**